report/sections/implementation.tex

```diff
@@ -1,19 +1,28 @@
-\section{Implementation}
+\section{Implementation} \label{impl}
 
-\todo{TODO: Explain each of the classifiers below and their hyper-parameters. Do not talk about their result, that belongs in the next section}
+% leave-one-out validation
+% k-fold validation
+% Holdout validation
+Once the features in the dataset had been cleansed and their dimensionality reduced, the k-fold, leave-one-out, and holdout validation methods were implemented to determine how to train the models. The holdout validation method performed best out of the three.
 
-\subsection{k-Nearest Neighbors (kNN)}
+% More on holdout validation
+% Solving class imbalance
+The holdout validation method consists of splitting the dataset in two independent sets: a train and a test set. The training set is used to train the model, while the test set is used to evaluate its performance. Since the dataset has class imbalance, a total of 60 samples ($\sim$$80\%$) per label  were randomly selected to be inserted in the training set, while the remaining samples ($\sim$$20\%$) were inserted in the test set. Hence, even though the dataset has class imbalance, the different models were trained using the same number of samples per label.
 
-\todo{TODO: What is kNN (short intro)?, hyper-parameters setup and what each of them mean}
+What follows is a short description of each of the models implemented and the hyper-parameters used to achieved the best results.
 
-\subsection{Support Vector Machine (SVM)}
+\subsection{kNN}
 
-\todo{TODO: What is a SVM (short intro)?, hyper-parameters setup and what each of them mean}
+A kNN classifier simply classifies samples by a majority vote of its neighbors, with a sample being assigned to the class most common among its \texttt{k} nearest neighbors \cite{data-mining-intro}. The hyper-parameter \texttt{n\_neighbors} determines the number of neighbors to use to choose the label of a sample, a total of 15 neighbors attained the best results.
 
-\subsection{Multilayer Perceptron (MLP)}
+\subsection{SVM}
 
-\todo{TODO: What is a MLP (short intro)?, hyper-parameters setup and what each of them mean}
+A SVM is a supervised learning algorithm which creates a hyperplane that represents the input samples as points in space, such that samples of different classes are separated from each other. New samples are then mapped into that same space and predicted to belong to a class based on where in that space these are located \cite{data-mining-intro}.
```

Comment:
I'm assuming we're spelling these out before this section?

Reply:
Yup, in the intro section